packages/core/src/parameters/getParameterDefinitionsFromSource.js

```diff
@@ -0,0 +1,175 @@
+      // group
+      const name = '_group_' + (groupIndex++)
+      const def = parseComment(code, lineNum, name)
+      let caption = def.caption
+      if (caption[0] === '>') {
+        caption = caption.substring(1).trim()
+        if (!def.options) def.options = {}
+        def.options.initial = 'closed'
+      }
+
+      defs.push(prev = { name, type: 'group', caption, ...def.options })
+    } else {
+      const idx = code.indexOf('/')
+      if (idx === -1) {
+        const def = parseDef(code, lineNum)
+        def.caption = def.name
+        defs.push(prev = def)
+      } else {
+        defs.push(prev = parseOne(
+          code.substring(idx).trim(),
+          code.substring(0, idx).trim(),
+          lineNum, lineNum
+        ))
+      }
+    }
+    i++
+  }
+
+  return defs
+}
+
+const parseOne = (comment, code, line1, line2) => {
+  let def = parseDef(code, line2)
+  const { caption, options } = parseComment(comment, line1, def.name)
+  def.caption = caption || def.name
+  if (options) {
+    def = { ...def, ...options }
+    if (def.type === 'checkbox' && 'initial' in def) def.checked = true
+    if (def.type === 'slider') {
+      if (def.min === undefined) {
+        def.min = 0
+      }
+      if (def.max === undefined) {
+        def.max = def.initial * 2 || 100
+      }
+    }
+  }
+
+  return def
+}
+
+const extractTextFromComment = (c) => {
+  const prefix = c.substring(0, 2)
+  if (prefix === '//') c = c.substring(2)
+  if (prefix === '/*') c = c.substring(2, c.length - 2)
+
+  return c.trim()
+}
+
+const parseComment = (comment, line, paramName) => {
+  comment = extractTextFromComment(comment)
+
+  const ret = {}
+  const idx = comment.indexOf('{')
+  if (idx !== -1) {
+    try {
+      // eslint-disable-next-line
+      ret.options = Function('return ' + comment.substring(idx)).call()
+    } catch (e) {
+      throw new EvalError(`${e.message}, parameter:${paramName}, line:${line}: ${comment.substring(idx)}`, 'code', line)
+    }
+    comment = comment.substring(0, idx).trim()
+  }
+
+  ret.caption = comment
+
+  return ret
+}
+
+const parseDef = (code, line) => {
+  if (code[code.length - 1] === ',') code = code.substring(0, code.length - 1).trim()
+  let idx = code.indexOf('=')
+
+  if (idx === -1) idx = code.indexOf(':')
+
+  if (idx === -1) {
+    return { name: code, type: 'text' }
+  } else {
+    const initial = code.substring(idx + 1).trim()
+
+    const ret = { type: 'text', name: code.substring(0, idx).trim() }
+
+    if (initial === 'true' || initial === 'false') {
+      ret.type = 'checkbox'
+      ret.checked = initial === 'true'
+    } else if (/^[0-9]+$/.test(initial)) {
+      ret.type = 'int'
+      ret.initial = parseFloat(initial)
+    } else if (/^[0-9]+\.[0-9]+$/.test(initial)) {
+      ret.type = 'number'
+      ret.initial = parseFloat(initial)
+    } else {
+      try {
+        // eslint-disable-next-line
+        ret.initial = Function('return ' + initial).bind({}).call()
```

Comment:
So... this is evaluating the variable definition in order to obtain the initial value. Why support anything but 'text' values?

I don't see why this should call 'return X'. That is really out of scope for these definitions. Can you provide some examples which make sense?

Reply:
I had a vague idea that in the future a more advanced input field in customizer could return and array of numbers, or an object. The parameters are not performance critical, it would be ok to only support strings, and any fancy input could always use JSON in the value.

Reply:
That sounds reasonable as custom parameter definitions should be possible in the future, hopefully.

Obviously, some nice documentation or example will be required later.

Reply:
moved to JSON5, so eval should not be a concern any more, and arrays or objects are supported as well